assets/javascripts/builder.jsx

```diff
@@ -0,0 +1,35 @@
+var React = require('react');
+var riek = require('riek')
+var RIEInput = riek.RIEInput;
+var FormElement = require('./formElement.jsx');
+
+module.exports = React.createClass({
+  render: function() {
+    var formElements = [];
+    var body;
+
+    for (var i = 0; i < this.props.formElements.length; i++) {
```

Comment:
Should I declare *var i *at the top of the whole file, or within the
*render* function for the React class?

On Fri, May 27, 2016 at 9:26 AM, Bookis Worthy <notifications@github.com>
wrote:

> In assets/javascripts/builder.jsx
> <https://github.com/smurfik/beacon/pull/12#discussion_r64931075>:
>
> > @@ -0,0 +1,35 @@
> > +var React = require('react');
> > +var riek = require('riek')
> > +var RIEInput = riek.RIEInput;
> > +var FormElement = require('./formElement.jsx');
> > +
> > +module.exports = React.createClass({
> > +  render: function() {
> > +    var formElements = [];
> > +    var body;
> > +
> > +    for (var i = 0; i < this.props.formElements.length; i++) {
>
> It's better to declare the var i at the top, then just for (i = 0... for
> the loop.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/smurfik/beacon/pull/12/files/47b8958491bff1639c004167eec3caf85550a6bc#r64931075>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AMlHyF0vc4jvZy1WiSvOcXDqOFlY1EGcks5qFxsbgaJpZM4Il6QV>
> .
>